go.mod

```diff
@@ -7,7 +7,7 @@ require (
 	github.com/mitchellh/go-homedir v1.1.0
 	github.com/newrelic/go-agent/v3 v3.20.3
 	github.com/newrelic/go-insights v1.0.3
-	github.com/newrelic/newrelic-client-go/v2 v2.21.1
+	github.com/newrelic/newrelic-client-go/v2 v2.21.2-0.20231011094303-bbae780fcced
```

Comment:
Please revert this change before you merge the PR, @pranav-new-relic 
This change was added to allow compilation/integration tests to succeed as they would need to point to the changes in the Go-Client counterpart of this PR.